Correção dos pontos comentados por Joathan nas Telas de (new) aliens,planets e powers.

Lucas-SilvaGit opened the pull request:
Esse PR tem objetivo de corrigir os pontos comentados por joathan visando melhorar o código.

![Screenshot_1](https://user-images.githubusercontent.com/20430922/135186083-aabad082-a87f-46cd-ae28-86558b5951cb.png)
![Screenshot_3](https://user-images.githubusercontent.com/20430922/135186095-009cd6cf-60eb-4c84-85b0-9da8dbbc72c2.png)
![Screenshot_4](https://user-images.githubusercontent.com/20430922/135186118-99ce64a8-67ad-437d-ad7a-c13114f25248.png)

Lucas-SilvaGit committed:
estilização dos formulários new

Lucas-SilvaGit committed:
Correção do código, adicionada a classe form para facilitar o acesso pelo appalications.scss para estilização

Lucas-SilvaGit committed:
Correção do código, adicionada a classe form para facilitar o acesso pelo appalications.scss para estilização

Lucas-SilvaGit committed:
Correção do código, adicionada a classe form para facilitar o acesso pelo appalications.scss para estilização

joathan commented on app/assets/stylesheets/application.scss:
```diff
@@ -67,4 +67,18 @@ body {
+  .container{
+    width: 30%;
+    height: 30%;
+  }
```
linha em branco...

Lucas-SilvaGit replied:
Linha em branco adicionado conforme solicitado.

joathan commented on app/assets/stylesheets/application.scss:
```diff
@@ -67,4 +67,18 @@ body {
\ No newline at end of file
+    display: block; }
+
+  form{  // classe dos formulários new
+    align-items: center;
+    display: flex; 
+    justify-content: center;
+    flex-direction: column;
+    width: 100%;
+    height: 100%;
+  }
```
todas essa `class` pode ser substituída por `class` do bootstrap.

Lucas-SilvaGit replied:
Corrigido

joathan commented on app/assets/stylesheets/application.scss:
```diff
@@ -67,4 +67,18 @@ body {
   label {
-    display: block; }
\ No newline at end of file
+    display: block; }
```
essa tbm...

Lucas-SilvaGit replied:
Corrigido

joathan commented on app/assets/stylesheets/application.scss:
```diff
@@ -67,4 +67,18 @@ body {
+    height: 100%;
+  }
+
+  .container{
+    width: 30%;
+    height: 30%;
```
aqui tbm....

Lucas-SilvaGit replied:
Corrigido

joathan commented on app/views/aliens/new.html.erb:
```diff
@@ -2,13 +2,10 @@
-    <br>
-    <br>
     
         <button type="button" class="btn btn-dark "><%= link_to 'Back', aliens_path %></button>
```
`<button type="button" class="btn btn-dark ">` não precisa.. é so por a `class` no `link_to`

Lucas-SilvaGit replied:
Corrigido

joathan replied:
não foi corrigido, continua...

joathan commented on app/views/planets/new.html.erb:
```diff
@@ -1,12 +1,11 @@
 <div class="container">
```
qual o motivo de usar essa `class` dentro da pagina?

Lucas-SilvaGit replied:
corrigido

joathan reviewed (changes requested):
revisa as considerações...

Lucas-SilvaGit committed:
adicionada linha em branco no final do arquivo

Lucas-SilvaGit committed:
adicionado kaminari

Lucas-SilvaGit committed:
nada

Lucas-SilvaGit committed:
removido todo css desnecessário

Lucas-SilvaGit committed:
removido container

Lucas-SilvaGit committed:
removido container

Lucas-SilvaGit committed:
removido container

Lucas-SilvaGit committed:
removido container

Lucas-SilvaGit committed:
foi adicionado as classes para tornar o container fluido no navbar e no yield do index dos aliens

Lucas-SilvaGit committed:
removido container

Lucas-SilvaGit committed:
removido container

Lucas-SilvaGit committed:
adicionado a classe card envolta da tabela, adicionada a classe flex nos elementos para melhorar o layout, feita alteração na estilização does botões

Lucas-SilvaGit committed:
criar pagina inicial

Lucas-SilvaGit committed:
arquivo de configuração kaminari

Lucas-SilvaGit committed:
removido container

Lucas-SilvaGit committed:
feita configuração para kaminari definindo o numero de registro por pagina

Lucas-SilvaGit committed:
foi definido o parametro page no controller de aliens porém não funcionou

Lucas-SilvaGit committed:
deletado

Lucas-SilvaGit committed:
deletado

Lucas-SilvaGit committed:
deletado

Lucas-SilvaGit committed:
deletado

Lucas-SilvaGit committed:
adicionado kaminari e bootstrap views

Lucas-SilvaGit committed:
teste

Lucas-SilvaGit committed:
adicionado os parametros para paginação

Lucas-SilvaGit committed:
adicionado os parametros para paginação

Lucas-SilvaGit committed:
adicionado os parametros para paginação

Lucas-SilvaGit committed:
adicionado os parametros para paginação

Lucas-SilvaGit committed:
nada foi mexido, só foi apagado e restaurado

Lucas-SilvaGit committed:
teste de imagem

joathan commented on app/controllers/aliens_controller.rb:
```diff
@@ -3,7 +3,7 @@ class AliensController < ApplicationController
   # GET /aliens or /aliens.json
   def index
-    @aliens = Alien.all
+    @aliens = Alien.all #page params[:page]
```
pq o comentario?

Lucas-SilvaGit replied:
veio no arquivo padrão
![image](https://user-images.githubusercontent.com/20430922/135556493-b01f9f48-24f6-4e49-8b56-d84598c1ac98.png)

joathan commented on app/models/alien_power.rb:
```diff
@@ -1,4 +1,6 @@
   belongs_to :alien
   belongs_to :power
+
+  #paginates_per 10
```
pq o comentario?

Lucas-SilvaGit replied:
ta comentado ai porque deve ter baixado antes do push
![image](https://user-images.githubusercontent.com/20430922/135556591-f2286da5-4d6c-4ecc-96f6-0b545e3b29b5.png)

joathan commented on app/views/Home/home-index.html.erb:
```diff
@@ -0,0 +1,4 @@
+<p id="notice"><%= notice %></p>
+
+<img src="C:\Ruby\Aliens\spacealiens\image" class="img-fluid" alt="...">
```
apos o deploy, esse caminho de pasta não vai existir no servidor... pesquise por [image_tag](https://apidock.com/rails/ActionView/Helpers/AssetTagHelper/image_tag)

Lucas-SilvaGit replied:
isso foi só um teste, que ia fazer mais não dei continuidade porque mudei de ideia

joathan replied:
no PR são coisas q vão pra master (produção).... 🤷🏿‍♂️

joathan commented on app/views/aliens/edit.html.erb:
```diff
@@ -1,4 +1,4 @@
-<div class="container">
+
     <div class="row align-items-center">
```
essa `div` abre, mas não fecha...

Lucas-SilvaGit replied:
seu arquivo ta diferente por algum motivo.

![image](https://user-images.githubusercontent.com/20430922/135556707-71d5370a-4fce-4ec5-935e-2fbb1979bf99.png)

Lucas-SilvaGit replied:
esse container removi ontem

joathan replied:
o `git` não tem como errar... ou ficou sem commitar, ou esta em outra branch... mas no PR não estava... eu falei da `div` não foi do `container` o `container` ja esta em vermelho.

joathan commented on app/views/layouts/application.html.erb:
```diff
@@ -11,24 +11,30 @@
+		  <div class="col-md-12">
+        <nav class="navbar navbar-expand-lg navbar-light bg-light static-top">
+          <a class="navbar-brand px-3" href="#">Space Aliens</a>
+          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
+            <span class="navbar-toggler-icon"></span>
+          </button>
+        
         <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
-        <div class="navbar-nav">
-          <a class="nav-link active" aria-current="page" href="#">Home</a>
-          <a class="nav-link" href="http://localhost:3000/aliens">Aliens</a>
-          <a class="nav-link" href="http://localhost:3000/planets">Planets</a>
-          <a class="nav-link" href="http://localhost:3000/powers">Powers</a>
+          <div class="navbar-nav">
+            <a class="nav-link active" aria-current="page" href="#">Home</a>
+            <a class="nav-link" href="http://localhost:3000/aliens">Aliens</a>
+            <a class="nav-link" href="http://localhost:3000/planets">Planets</a>
+            <a class="nav-link" href="http://localhost:3000/powers">Powers</a>
+          </div>
         </div>
+        </nav>
```
o ideal é q essa `nav` fique em uma `partial` mas por hora, deixa assim....

Lucas-SilvaGit replied:
certo

joathan commented on app/views/planets/edit.html.erb:
```diff
@@ -1,4 +1,4 @@
-<div class="container">
+<
```
pra q esse `<` ?

Lucas-SilvaGit replied:
removido, vestígio de código mau apagado

joathan commented on app/views/planets/edit.html.erb:
```diff
@@ -1,4 +1,4 @@
-<div class="container">
+<
     <div class="row align-items-center">
```
essa `div` abre mais não fecha..

Lucas-SilvaGit replied:
seu código esta diferente. 
todos os containers foram removidos do código no commit de ontem

joathan replied:
estou pelo codigo do PR...

joathan commented on app/views/planets/edit.html.erb:
```diff
@@ -10,4 +10,3 @@
         
         <button type="button" class="btn btn-dark "><%= link_to 'Show', @planet %></button>
         <button type="button" class="btn btn-dark "><%= link_to 'Back', planets_path %></button>
-</div>
```
faltou linha em branco..

Lucas-SilvaGit replied:
corrigido

joathan replied:
todo commit ele gera um hash. copia aqui em qual commit ele foi corrigido por favor.

joathan commented on app/views/planets/new.html.erb:
```diff
@@ -1,12 +1,10 @@
 
         <button type="button" class="btn btn-dark "><%= link_to 'Back', planets_path %></button>
     </div>
-</div>
```
falta a linha

Lucas-SilvaGit replied:
corrigido
![image](https://user-images.githubusercontent.com/20430922/135557242-912794ab-0edc-49d4-8a28-068a70619f50.png)

joathan replied:
todo commit ele gera um hash. copia aqui em qual commit ele foi corrigido por favor.

joathan commented on app/views/powers/edit.html.erb:
```diff
@@ -1,4 +1,4 @@
-<div class="container">
+
```
pra que essa linha em branco?

Lucas-SilvaGit replied:
Corrigido

joathan commented on Gemfile:
```diff
@@ -21,6 +21,9 @@ gem 'jbuilder', '~> 2.7'
 # Use Active Model has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
+gem 'kaminari'
+gem 'bootstrap4-kaminari-views'
+gem 'bootstrap', '~> 4.3.1'
```
não precisa essas `gems`

Lucas-SilvaGit replied:
certo

joathan commented on Gemfile:
```diff
@@ -21,6 +21,9 @@ gem 'jbuilder', '~> 2.7'
 # gem 'redis', '~> 4.0'
 # Use Active Model has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
+gem 'kaminari'
```
use assim:
`gem 'kaminari', :git => 'https://github.com/kaminari/kaminari'`

Lucas-SilvaGit replied:
funcionou!

Lucas-SilvaGit committed:
Resolvido conflitos

- Tradução dos nomes do menu do navbar
- Alteração no caminho da imagem, e tradução dos nomes do menu
